src/hardware/mouse/mouse_interfaces.cpp

```diff
@@ -0,0 +1,818 @@
+/*
+ *  Copyright (C) 2022-2022  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "mouse_interfaces.h"
+#include "mouse.h"
+#include "mouse_common.h"
+#include "mouse_config.h"
+#include "mouse_manymouse.h"
+#include "mouse_queue.h"
+
+#include "checks.h"
+
+CHECK_NARROWING();
+
+static MouseQueue &mouse_queue = MouseQueue::GetInstance();
+
+std::vector<MouseInterface *> mouse_interfaces = {};
+
+// ***************************************************************************
+// Mouse interface information facade
+// ***************************************************************************
+
+MouseInterfaceInfoEntry::MouseInterfaceInfoEntry(const MouseInterfaceId interface_id)
+        : idx(static_cast<uint8_t>(interface_id))
+{}
+
+const MouseInterface &MouseInterfaceInfoEntry::Interface() const
+{
+	return *mouse_interfaces[idx];
+}
+
+const MousePhysical &MouseInterfaceInfoEntry::MappedPhysical() const
+{
+	const auto idx = Interface().GetMappedDeviceIdx();
```

Comment:
`idx` is also a member of this `MouseInterfaceInfoEntry` class.
Should we update it, or is this a temporary?
(If it's a temporary, suggest a different name to avoid shadowing)

Reply:
No, this is just a temporary variable. Changed both names to avoid confusion.